code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -7453,6 +7453,37 @@ var/global/list/bomb_like_items = list(/obj/item/device/transfer_valve, /obj/ite
 	reagents.add_reagent(ZAMSPICYTOXIN, 5)
 	bitesize = 3
 
+/obj/item/weapon/reagent_containers/food/snacks/blethernoodlesoup
+	name = "Blether Noodle Soup"
+	desc = "A hearty grey noodle soup. Great for teaching growing greylings new words! Not to be confused with human alphabet soup."
+	icon_state = "blethernoodlesoup_open"
+	trash = /obj/item/weapon/reagent_containers/glass/soupcan
+	food_flags = FOOD_MEAT | FOOD_LIQUID
+	crumb_icon = "dribbles"
+	filling_color = "#FF9700"
+	valid_utensils = UTENSILE_FORK|UTENSILE_SPOON
+	bitesize = 3
+	wrapped = FALSE
+
+/obj/item/weapon/reagent_containers/food/snacks/blethernoodlesoup/wrapped
+	icon_state = "blethernoodlesoup_closed"
+	wrapped = TRUE
+
+/obj/item/weapon/reagent_containers/food/snacks/blethernoodlesoup/attack_self(mob/user)
+	if(wrapped)
+		Unwrap(user)
+		playsound(user, 'sound/effects/can_open1.ogg', 50, 1)
+		reagents.add_reagent(NUTRIMENT, 6)
+		reagents.add_reagent(SACID, 10)
+		reagents.add_reagent(LOCUTOGEN, 5)
+	else
+		..()
+
+/obj/item/weapon/reagent_containers/food/snacks/blethernoodlesoup/proc/Unwrap(mob/user)
+	icon_state = "blethernoodlesoup_open"
+	to_chat(user, "<span class='notice'>You pull the tab on the soup can and pop the lid open. An inviting smell wafts out.")
```

Comment:
add a sanity check here for user
```
if(user)
   to_chat(user, ...
 ```
 
The reason for this is mainly if an admin unwraps it with a proccall

Reply:
Had some difficulty with this one, but hopefully I managed to do it.